### PR TITLE
Improve request_options

### PR DESCRIPTION
**Pull request by MarkDaoust**
Fixes: https://github.com/google-gemini/generative-ai-python/issues/223

### CODE REVIEW

File: google/generativeai/types/helper_types.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        self.timeout = timeout
+
+    # Inherit from Mapping for **unpacking
+    def __getitem__(self, item):
+        if item == "retry":
+            return self.retry
+        elif item == "timeout":
+            return self.timeout
+        else:
+            raise KeyError(f'RequestOptions does not have a "{item}" key')
```
Comment (mayureshagashe2105) [inline google/generativeai/types/helper_types.py:74]: getattr(self, item)  # maybe?

**Review by mayureshagashe2105 — APPROVED** [review]

File: google/generativeai/types/helper_types.py (hunk 1 above)
Comment (MarkDaoust) [reply google/generativeai/types/helper_types.py:74]: Fewer lines of code sounds good. 

But then that works for "_" attributes? and what lookups succeed don't match a contains check? :/

Comment (mayureshagashe2105) [reply google/generativeai/types/helper_types.py:74]: I see, "_" shouldn't be accessible.
LGTM

Comment (markmcd) [reply google/generativeai/types/helper_types.py:74]: You could do this with `self.__dataclass_fields__.keys()`, but repetition is required in this file anyway.

File: tests/test_discuss.py [hunk 2]
```diff
@@ -21,6 +21,7 @@
 
 from google.generativeai import discuss
 from google.generativeai import client
+from google.generativeai import types
```
Comment (markmcd) [inline tests/test_discuss.py:24]:
```suggestion
```

Is this needed? looks like it's unused in the file.

File: google/generativeai/answer.py [hunk 3]
```diff
@@ -247,7 +245,7 @@ def generate_answer(
     safety_settings: safety_types.SafetySettingOptions | None = None,
     temperature: float | None = None,
     client: glm.GenerativeServiceClient | None = None,
-    request_options: dict[str, Any] | None = None,
+    request_options: helper_types.RequestOptionsType | None = None,
```
Comment (markmcd) [inline google/generativeai/answer.py:248]: Just an FYI that we might need to make edits when this is imported to google3 too - e.g. cl/625239248.

**Review by markmcd — APPROVED** [review]
Definitely not blocking, but it'd be nice if both the dict & DC params were tested explicitly, rather than as part of the feature tests (e.g. something like test_helpers.py with test_request_options_dict, test_request_options_dc)

File: google/generativeai/answer.py (hunk 3 above)
Comment (MarkDaoust) [reply google/generativeai/answer.py:248]: Ack.

**Review by markmcd — APPROVED** [review]

**Pull request merged**